### PR TITLE
ci/package_checks: Add check for order of `{build,check,run}deps`

### PR DESCRIPTION
**Pull request by silkeh**
**Summary**

This adds a CI check for the order of build dependencies.

**Test Plan**

See [this test PR](https://github.com/silkeh/packages/pull/3/files).

**Checklist**

- ~~Package was built and tested against unstable~~

### CODE REVIEW

**Comment by joebonrichie** [comment]
Oooh, nice work! Needs to check the newly added `checkdeps` aswell

**Comment by silkeh** [comment]
@joebonrichie: good point, I added `rundeps` as well.

File: common/CI/package_checks.py [hunk 1]
```diff
@@ -304,6 +342,7 @@ def _commit_package_yaml(self, ref: str) -> Optional[Dict[str, Any]]:
 class Checker:
     checks = [
         Homepage,
+        PackageBuildDepsOrder,
```
Comment (joebonrichie) [inline common/CI/package_checks.py:345]: nit: maybe dependency orders or something similar as it checks all dep types?

Comment (silkeh) [reply common/CI/package_checks.py]: Made it `PackageDependenciesOrder`.

**Review by joebonrichie — APPROVED** [review]

**Pull request merged**